src/api/app/services/architectures_controller_service/architecture_updater.rb

```diff
@@ -1,21 +1,22 @@
 module ArchitecturesControllerService
   class ArchitectureUpdater
-    def initialize(archs)
-      @archs = archs
-      @all_valid = true
+    def initialize(params)
+      @archs = params.require(:archs).permit!
+      @all_valid = false
     end
 
     def call
-      @archs.each do |name, value|
+      @all_valid = @archs.to_h.map do |name, value|
```

Comment:
`to.h` + `map` is a not an efficient thing and you store objects instead of just a boolean in `@all_valid`... (you save one query to the database in exchange but...) as you say this efficiency is not important here, let's say this is a nitpick... but I still don't understand the benefit of the refactoring... (sorry @Ana06 of the past for [the contradiction](https://github.com/openSUSE/open-build-service/pull/6714#discussion_r246697320))

[for the @Ana06's of the future] I see a nice refactoring here once we update to Ruby 2.6 :full_moon: